app/models/redemption.rb

```diff
@@ -0,0 +1,19 @@
+# Redemption class
+# redemptions store the event of a user redeeming a voucher
+
+class Redemption < ApplicationRecord
+  belongs_to :voucher
+  belongs_to :user
+  has_many :claims, dependent: :destroy
+  has_many :claimed_tutorials, through: :claims, source: :claimable,
+                               source_type: "Tutorial"
+  has_many :claimed_talks, through: :claims, source: :claimable,
+                           source_type: "Talk"
+
+  delegate :lecture, to: :voucher
+  delegate :role, to: :voucher
+  delegate :tutor?, to: :voucher
+  delegate :editor?, to: :voucher
+  delegate :teacher?, to: :voucher
+  delegate :speaker?, to: :voucher
```

Comment:
This delegates to methods that stem from the definition of an enum in the `Voucher` class. It complicates the code path and adds one additional layer that is unnecessary IMHO. What is the advantage of having `redemption.editor?` over `redemption.voucher.editor?` (or something similar)

Reply:
I do not cling to the delegate methods. It is a Rails Active Support Core extension, see [here](https://guides.rubyonrails.org/active_support_core_extensions.html#delegate). As usual, there a are blog posts pro and contra delegation. If you think we should not use it, I have no problem when we delete these delegations.

Reply:
Removed in e2786e3f49